test/test_reader.py

```diff
@@ -57,3 +69,27 @@ def test_value_error_on_invalid_dimensionality(self, dims, min_reader_data):
         with pytest.raises(ValueError):
             create_reader(min_reader_data,
                           dimensionality=dims)
+
+
+@given(dataset(dims=2))
```

Comment:
Does this have to be two, or could it be an integers strategy?

Reply:
I expect that we'll evolve this eventually to test all meaningful dimensionalities. I don't know if an integer strategy is the best way to go, though. From what I can see, substantially different things happen internally depending on dimensionality, so it might be best to explicitly test 1, 2, and 3 (and more?) dimensions.

Reply:
Yes, separate tests makes more sense.